prototypes/technology/trainassembly-technology.lua

```diff
@@ -0,0 +1,151 @@
+require "util"
+
+local trainTech = util.table.deepcopy(data.raw["technology"]["railway"])
+
+  trainTech.name = "trainassembly-trainTechnology"
+  trainTech.effects = {}
+  trainTech.localised_name = {"technology-name.trainTech"}
+  trainTech.localised_description = {"technology-description.trainTech"}
+
+  data:extend({
+    trainTech,
+  })
+
+for _, trainTechn in pairs ({
+  "locomotive",
+  "cargo-wagon",
+}) do
+  table.insert(data.raw["technology"][trainTech.name].effects,
+  {
+    type = "unlock-recipe",
+    recipe = trainTechn,
+  })
+end
+
+
+if data.raw["technology"]["railway"].effects then
+  for effectIndex, effect in pairs(data.raw["technology"]["railway"].effects) do
+    if effect.type == "unlock-recipe"
+    and (effect.recipe == "locomotive" or effect.recipe == "cargo-wagon") then
+      data.raw["technology"]["railway"].effects[effectIndex] = nil -- this removes that single unlock
+    end
+  end
+
+else
+  data.raw["technology"]["railway"].effects = {}
+end
+
+
+table.insert(data.raw["technology"]["railway"].effects,
+{
+  type = "unlock-recipe",
+  recipe = "trainassembly-trainfuel-raw-wood",
```

Comment:
I would switch the inserts of the trainfuel and the trainassembler so the technology shows rails, building and then the fuel (the order of inserting it will be the order of showing the unlocks ingame)